src/main/java/com/deuna/maven/InitElements.kt

```diff
@@ -26,39 +33,44 @@ fun DeunaSDK.initElements(
     closeEvents: Set<ElementsEvent> = emptySet(),
     userToken: String? = null,
     userInfo: UserInfo? = null,
-    cssFile: String? = null
+    cssFile: String? = null,
+    types: List<Json> = emptyList(),
 ) {
     val baseUrl = this.environment.elementsBaseUrl
 
     ElementsActivity.setCallbacks(sdkInstanceId = sdkInstanceId, callbacks = callbacks)
 
-    val queryParameters = mutableMapOf<String, String>()
-    queryParameters[QueryParameters.MODE.value] = QueryParameters.WIDGET.value
-    queryParameters[QueryParameters.PUBLIC_API_KEY.value] = publicApiKey
+    val queryParameters = mutableMapOf(
+        QueryParameters.MODE.value to QueryParameters.WIDGET.value,
+        QueryParameters.PUBLIC_API_KEY.value to publicApiKey
+    )
 
-    if (!userToken.isNullOrEmpty()) {
-        queryParameters[QueryParameters.USER_TOKEN.value] = userToken
-    } else if (userInfo != null) {
-        if (!userInfo.isValid()) {
-            // if the user token is not passed or is empty the userInfo must be passed
-            DeunaLogs.error(ElementsErrorMessages.INVALID_USER_INFO.message)
-            callbacks.onError?.invoke(ElementsErrors.invalidUserInfo)
+    when {
+        !userToken.isNullOrEmpty() -> queryParameters[QueryParameters.USER_TOKEN.value] = userToken
+        userInfo != null && userInfo.isValid() -> {
+            queryParameters.apply {
+                put(QueryParameters.FIRST_NAME.value, userInfo.firstName)
+                put(QueryParameters.LAST_NAME.value, userInfo.lastName)
+                put(QueryParameters.EMAIL.value, userInfo.email)
+            }
+        }
+
+        else -> {
+            DeunaLogs.error(ElementsErrorMessages.MISSING_USER_TOKEN_OR_USER_INFO.message)
+            callbacks.onError?.invoke(ElementsErrors.missingUserTokenOrUserInfo)
             return
         }
-        queryParameters[QueryParameters.FIRST_NAME.value] = userInfo.firstName
-        queryParameters[QueryParameters.LAST_NAME.value] = userInfo.lastName
-        queryParameters[QueryParameters.EMAIL.value] = userInfo.email
-    } else {
-        // if the user token is not passed or is empty the userInfo must be passed
-        DeunaLogs.error(ElementsErrorMessages.MISSING_USER_TOKEN_OR_USER_INFO.message)
-        callbacks.onError?.invoke(ElementsErrors.missingUserTokenOrUserInfo)
     }
 
-    if (!cssFile.isNullOrEmpty()) {
-        queryParameters[QueryParameters.CSS_FILE.value] = cssFile
+    cssFile?.let {
+        queryParameters[QueryParameters.CSS_FILE.value] = it
```

Comment:
@darwinmorocho-deuna porque `it` ? No es una variable que se entienda

Reply:
en kotlin para evitar el uso de un if para validar que un dato sea nulo o vacio se pude usar let , por defecto usa it, pero se lo puede cambiar solo que lo vi un poco redundante hacer algo como

```
    cssFile?.let { cssFile
        queryParameters[QueryParameters.CSS_FILE.value] = cssFile
        }
```